structured/structured.proto

```diff
@@ -79,9 +79,11 @@ message TableDescriptor {
   // next_column_id is used to ensure that deleted column ids are not reused.
   optional uint32 next_column_id = 4 [(gogoproto.nullable) = false,
       (gogoproto.customname) = "NextColumnID"];
-  repeated IndexDescriptor indexes = 5 [(gogoproto.nullable) = false];
+  optional IndexDescriptor primary_index = 5 [(gogoproto.nullable) = false];
```

Comment:
hm. Perhaps this should not be `nullable = false` since that causes awkward checks like `desc.PrimaryIndex.Name != ""` rather than `desc.PrimaryIndex != nil`

Also, are we calling it a primary index or a primary key?